tests/test_functions.py

```diff
@@ -70,16 +71,39 @@ def test_exp() -> None:
 
 def test_log() -> None:
     expr = psi * X(4)
-    assert log(expr) == Expression(
-        Expression.Tag.FN,
-        Expression.symbol("log"),
-        Expression.mul(
-            Expression.symbol("psi", trainable=True),
-            Expression.quantum_operator(
-                Expression.symbol("X"),
-                Support(4),
-                is_hermitian=True,
-                is_unitary=True,
+    assert log(expr) == Expression.quantum_operator(
```

Comment:
I was also surprised the `loq` would return a function expression as opposed to a quantum op. However, the previous version of the test is also succeeding. Isn't that an issue ?

Reply:
Not sure if I understood it correctly.

When I run the previous test against the current version of `log`, it returns the following error.
```
FAILED tests/test_functions.py::test_log - AssertionError: assert QuantumOperator(Function(Symbol('log'), Multiply(Symbol('psi', trainable=True), QuantumOperator(Symbol('X'), [4], is_hermitian=True, is_unitary=True))), [4]) == Function(Symbol('log'), Multiply(Symbol('psi', trainable=True), QuantumOperator(Symbol('X'), [4], is_hermitian=True, is_unitary=True)))
  
  Full diff:
  - Function(Symbol('log'), Multiply(Symbol('psi', trainable=True), QuantumOperator(Symbol('X'), [4], is_hermitian=True, is_unitary=True)))
  + QuantumOperator(Function(Symbol('log'), Multiply(Symbol('psi', trainable=True), QuantumOperator(Symbol('X'), [4], is_hermitian=True, is_unitary=True))), [4])
  ? ++++++++++++++++                                                                                                                                       ++++++
```

But it should pass if the expression _does not contain_ any quantum operator.

Reply:
Ah sorry, please ignore me I confused myself with smtg else. Thanks.

Reply:
No worries!